Objects/exceptions.c

```diff
@@ -2929,9 +2929,12 @@ _check_for_legacy_statements(PySyntaxErrorObject *self, Py_ssize_t start)
             return -1;
         }
     }
-    if (PyUnicode_Tailmatch(self->text, print_prefix,
-                            start, text_len, -1)) {
-
+    Py_ssize_t match = PyUnicode_Tailmatch(self->text, print_prefix,
+                                           start, text_len, -1);
+    if (match == -1) {
+        return -1;
+    }
+    if (match == 1) {
```

Comment:
Since `match` is a boolean here, I prefer to use just `if (match)` instead of `if (match == 1)`.